### PR TITLE
(PDB-3630) add support for facts blacklist in PDB config 4.4.x

### PR DESCRIPTION
**Pull request by Zak-Kent**
* adds facts blacklist option to PDB user config

### CODE REVIEW

**Comment by Zak-Kent** [comment]
@puppetlabs-jenkins retest this please

File: src/puppetlabs/puppetdb/command.clj [hunk 1]
```diff
@@ -504,7 +510,8 @@
          [(global-metric :processing-time)
           (cmd-metric command version :processing-time)]
 
-         (process-command-and-respond! cmd scf-write-db response-chan stats)
+         ;; added facts blacklist here, delete me
```
Comment (mullr) [inline src/puppetlabs/puppetdb/command.clj]: stray comment?

File: src/puppetlabs/puppetdb/config.clj [hunk 2]
```diff
@@ -71,7 +71,8 @@
      :stats (pls/defaulted-maybe String "true")
      :log-statements (pls/defaulted-maybe String "true")
      :statements-cache-size (pls/defaulted-maybe s/Int 0)
-     :connection-timeout (pls/defaulted-maybe s/Int 3000)}))
+     :connection-timeout (pls/defaulted-maybe s/Int 3000)
+     :facts-blacklist String}))
```
Comment (mullr) [inline src/puppetlabs/puppetdb/config.clj]: We should allow the user to provide either strings of comma-separated fact names or an array of fact names; the latter can come from a .conf (HOCON) configuration file.

File: test/puppetlabs/puppetdb/config_test.clj [hunk 3]
```diff
@@ -109,7 +109,16 @@
                        (configure-section :database write-database-config-in write-database-config-out)
                        configure-read-db)]
         (is (= (get-in config [:read-database :maximum-pool-size]) 25))
-        (is (= (get-in config [:database :maximum-pool-size]) 25))))))
+        (is (= (get-in config [:database :maximum-pool-size]) 25))))
+
+    (testing "facts-blacklist string is converted correctly"
```
Comment (mullr) [inline test/puppetlabs/puppetdb/config_test.clj]: need a similar test to see that ["fact1" "fact2"] is left alone.

**Pull request merged**